src/Raven.Server/Documents/Sharding/Handlers/ShardedIncomingReplicationHandler.cs

```diff
@@ -190,42 +208,76 @@ private ReplicationBatch[] PrepareReplicationDataForShards(TransactionOperationC
             {
                 foreach (var attachment in dataForReplicationCommand.ReplicatedAttachmentStreams)
                 {
+                    DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Shiran, DevelopmentHelper.Severity.Normal, "Make '_tempFileCache.MemoryStreamCapacity' configurable");
+                    var stream = attachment.Value.Size > _tempFileCache.MemoryStreamCapacity
+                            ? _tempFileCache.RentFileStream()
+                            : _tempFileCache.RentMemoryStream();
+
+                    _batches.Streams.Add(attachment.Key, stream);
+
+                    attachment.Value.Stream.Seek(0, SeekOrigin.Begin);
+                    attachment.Value.Stream.CopyTo(stream);
+
                     for (var shard = 0; shard < _parent.Context.ShardCount; shard++)
                     {
                         if (batches[shard].Attachments?.ContainsKey(attachment.Key) != true)
                             continue;
 
-                        var buffer = GetBufferFromAttachmentStream(attachment.Value.Stream);
-
-                        var attachmentStream = new AttachmentReplicationItem
+                        AddAttachmentStreamItemToBatch(batches[shard], attachment.Key, stream);
```

Comment:
after we added this, can we dispose the `attachment`?

Reply:
we are disposing it later in `HandleBatchAsync`